### ISSUE TITLE
mysql_upgrade after update

### ISSUE THREAD

**Issue by AikoAppeldorn** [mailcow/mailcow-dockerized#2092]
**Bug**
After the latest update the mysql-mailcow docker is always restarting and has following warning:
`Warning] InnoDB: Table mysql/innodb_index_stats has length mismatch in the column name table_name.  Please run mysql_upgrade`

**Expected behavior**
According to mailcow.email the mysql_upgrade could be call via the docker api.
The php-fpm-mailcow calls the mysql_upgrade task, but then the mysql went down and it starts from the beginning.

**Logs**
[mysql-mailcow.txt](https://github.com/mailcow/mailcow-dockerized/files/2677265/mysql-mailcow.txt)
[php-fpm-mailcow.txt](https://github.com/mailcow/mailcow-dockerized/files/2677253/php-fpm-mailcow.txt)

**Comment by andryyy** [comment]
You should recheck all images are up-to-date (esp dockerapi-mailcow) and maybe let it print the returned lines, when it runs mysql_upgrade. You may also also run mysql_upgrade manually via docker exec and check the result, probably easier. Sounds like it just fails to upgrade.

**Comment by AikoAppeldorn** [comment]
Thank you for yours answer.

The docker images are all up to date.
Output from the mysql_ugprade:
```
2018-12-13 19:56:41 139822526101248 [Note] InnoDB: Buffer pool(s) load completed at 181213 19:56:41
Phase 1/7: Checking and upgrading mysql database
Processing databases
mysql
mysql.column_stats                                 OK
mysql.columns_priv                                 OK
mysql.db                                           OK
mysql.event                                        OK
mysql.func                                         OK
mysql.gtid_slave_pos                               OK
mysql.help_category                                OK
mysql.help_keyword                                 OK
mysql.help_relation                                OK
mysql.help_topic                                   OK
mysql.host                                         OK
mysql.index_stats                                  OK
mysql.innodb_index_stats                           OK
mysql.innodb_table_stats                           OK
mysql.plugin                                       OK
mysql.proc                                         OK
mysql.procs_priv                                   OK
mysql.proxies_priv                                 OK
mysql.roles_mapping                                OK
mysql.servers                                      OK
mysql.table_stats                                  OK
mysql.tables_priv                                  OK
mysql.time_zone                                    OK
mysql.time_zone_leap_second                        OK
mysql.time_zone_name                               OK
mysql.time_zone_transition                         OK
mysql.time_zone_transition_type                    OK
mysql.user                                         OK
Phase 2/7: Installing used storage engines... Skipped
Phase 3/7: Fixing views
2018-12-13 19:56:50 139823188268800 [Warning] InnoDB: Table mysql/innodb_table_stats has length mismatch in the column name table_name.  Please run mysql_upgrade
2018-12-13 19:56:50 139823188268800 [Warning] InnoDB: Table mysql/innodb_index_stats has length mismatch in the column name table_name.  Please run mysql_upgrade
2018-12-13 19:56:50 139823188268800 [Warning] InnoDB: Table mysql/innodb_table_stats has length mismatch in the column name table_name.  Please run mysql_upgrade
2018-12-13 19:56:50 139823188268800 [Warning] InnoDB: Table mysql/innodb_index_stats has length mismatch in the column name table_name.  Please run mysql_upgrade
2018-12-13 19:56:50 139823188268800 [Warning] InnoDB: Table mysql/innodb_table_stats has length mismatch in the column name table_name.  Please run mysql_upgrade
2018-12-13 19:56:50 139823188268800 [Warning] InnoDB: Table mysql/innodb_index_stats has length mismatch in the column name table_name.  Please run mysql_upgrade
2018-12-13 19:56:50 139823188268800 [Warning] InnoDB: Table mysql/innodb_table_stats has length mismatch in the column name table_name.  Please run mysql_upgrade
2018-12-13 19:56:50 139823188268800 [Warning] InnoDB: Table mysql/innodb_index_stats has length mismatch in the column name table_name.  Please run mysql_upgrade
2018-12-13 19:56:50 139823188268800 [Warning] InnoDB: Table mysql/innodb_table_stats has length mismatch in the column name table_name.  Please run mysql_upgrade
2018-12-13 19:56:50 139823188268800 [Warning] InnoDB: Table mysql/innodb_index_stats has length mismatch in the column name table_name.  Please run mysql_upgrade
mailcow.grouped_domain_alias_address               OK
mailcow.grouped_mail_aliases                       OK
mailcow.grouped_sender_acl                         OK
mailcow.sieve_after                                OK
mailcow.sieve_before                               OK
mailcow.sogo_view                                  OK
Phase 4/7: Running 'mysql_fix_privilege_tables'
2018-12-13 19:56:51 139823065052928 [Note] Event Scheduler: scheduler thread started with id 15
2018-12-13 19:56:51 139823188268800 [Warning] InnoDB: Table mysql/innodb_table_stats has length mismatch in the column name table_name.  Please run mysql_upgrade
2018-12-13 19:56:51 139823188268800 [Warning] InnoDB: Table mysql/innodb_table_stats has length mismatch in the column name table_name.  Please run mysql_upgrade
Phase 5/7: Fixing table and database names
Phase 6/7: Checking and upgrading tables
Processing databases
information_schema
mailcow
mailcow._sogo_static_view                          OK
mailcow.admin                                      OK
mailcow.alias                                      OK
mailcow.alias_domain                               OK
mailcow.api                                        OK
mailcow.bcc_maps                                   OK
mailcow.da_acl                                     OK
mailcow.domain                                     OK
mailcow.domain_admins                              OK
mailcow.filterconf                                 OK
mailcow.forwarding_hosts                           OK
mailcow.imapsync                                   OK
mailcow.logs                                       OK
mailcow.mailbox                                    OK
mailcow.quarantine                                 OK
mailcow.quota2                                     OK
mailcow.recipient_maps                             OK
mailcow.relayhosts                                 OK
mailcow.sender_acl                                 OK
mailcow.settingsmap                                OK
mailcow.sieve_filters                              OK
mailcow.sogo_acl                                   OK
mailcow.sogo_alarms_folder                         OK
mailcow.sogo_cache_folder                          OK
mailcow.sogo_folder_info                           OK
mailcow.sogo_quick_appointment                     OK
mailcow.sogo_quick_contact                         OK
mailcow.sogo_sessions_folder                       OK
mailcow.sogo_store                                 OK
mailcow.sogo_user_profile                          OK
mailcow.spamalias                                  OK
mailcow.tfa                                        OK
mailcow.tls_policy_override                        OK
mailcow.user_acl                                   OK
mailcow.versions                                   OK
performance_schema
Phase 7/7: Running 'FLUSH PRIVILEGES'
2018-12-13 19:56:51 139823188268800 [Warning] 'proxies_priv' entry '@% root@' ignored in --skip-name-resolve mode.

```
The mysql is restarting, anyway.

**Comment by andryyy** [comment]
It was important to see why it fails. Try to drop the view sogo_view before running the upgrade.

**Comment by AikoAppeldorn** [comment]
I dropped the view sogo_view and then I run the mysql_upgrade.
But the log is the same and there is no more output.

**Comment by AikoAppeldorn** [comment]
In https://github.com/mailcow/mailcow-dockerized/blob/master/data/Dockerfiles/phpfpm/docker-entrypoint.sh#L29 is a check if there is a warning and I get always this warning: `2018-12-13 19:56:51 139823188268800 [Warning] 'proxies_priv' entry '@% root@' ignored in --skip-name-resolve mode. `

Is that maybe the problem?

**Comment by AikoAppeldorn** [comment]
Yes, I removed skip-name-resolve for testing and now the mysql-mailcow docker is running.

**Comment by andryyy** [comment]
That's not checked and not a problem for mysql_upgrade. It just didn't upgrade the tables, probably because of that broken view.

**Comment by andryyy** [comment]
We should drop the views before anything else. Or do we? I will check it.

**Comment by AikoAppeldorn** [comment]
> That's not checked and not a problem for mysql_upgrade. It just didn't upgrade the tables, probably because of that broken view.

Ok, now it works. Thank you for your help.

> We should drop the views before anything else. Or do we? I will check it.
👍

**Comment by andryyy** [comment]
Kind of fixed with 468e3dbe12884bac567834cae7b44e44210ea61b
